caches/infinispan/src/main/java/org/commonjava/maven/galley/cache/infinispan/FastLocalCacheProvider.java

```diff
@@ -352,6 +361,10 @@ else if ( e instanceof InterruptedException )
             {
                 while ( !canStreamOpen.get() )
                 {
+                    if ( copyExceOccurs.get() )
+                    {
+                        return null;
+                    }
                     try
                     {
                         copyLock.wait();
```

Comment:
This wait needs to have a timeout on it so the while condition can be checked continuously. I'd set it around 100ms.

Reply:
I think this does not need a timeout. The main thread waits here until it is waked up by the copy thread (when either read-ready or copy-error). The while just makes sure it go to right path after it is waked up. We don't need a continuous check here.